### DISCUSSION TITLE
FR: Allow batch select/actions on fields in Layout Designer

### DISCUSSION THREAD

**Discussion by ldstevens** [craftcms/cms#8429] [Ideas]
Quick feature request :) 

When putting together complex field layouts, it's a pain to drag & drop fields one at a time when picking fields out of 60-odd spread across half a dozen groups, especially when the groups wrap below other (long) groups, making the drag target far away.

Instead I wonder if having checkboxes with those fields would allow you to just run through all fields, check the ones you want, and then perform a batch action of "Add all to [tab x]", in a similar way to how you can select multiple entries and then perform a batch action that way. That way you can "shop" for your fields, and it only takes one click per field to add (+ batch action).

You could do the same thing with fields already added to a tab to batch remove them, or batch move them to another tab, etc.

Cheers :)

(Ah I see similar ideas have come up before: https://github.com/craftcms/cms/issues/1076 )

**Comment by AbbeyDesign** [comment]
After about 20-30 fields it gets a bit unwieldy. The Freeform form plugin has a nice approach to the way they manage their fields. Maybe something could be gleaned from their approach.

**Comment by dougstjohn** [comment]
Two ideas to improve this UI:
- Add one or more 'search' fields that filter by field name, field handle, or field type.
- Right click, double click or click a gear on any field and pick which tab you want to add the field to.

I also like the idea of selecting multiple fields and dragging them to a tab.

**Comment by eddiedale** [comment]
I think some small tweaks would go a long way here.

1. I think the sidebar should be on the left side, instead of right. At least in my experience, its more often fewer columns than many, and in the case of fewer, its less distance for drag-and-dropping from the left side.
2. It would be great to be able to drag an entire field group in. Optimally it would then just create a tab for this whole group if not dragged into a specific group (Do I remember wrong, or was this actually the case in a previous version of Craft?)

**Comment by beckbailey** [comment]
We frequently set up the same type of sections and would love to see some improvements made here. Being able to add an entire field group would be a great start.

**Reply by brandonkelly** [reply]
I suspect that won’t be as common in Craft 5, where entry types can be reused across multiple sections (combined with conditional fields added in Craft 4).
